crates/sui-adapter-transactional-tests/tests/dev_inspect/reference_coin_op_writes_valid.move

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// tests valid writes of mut references using coin operations
```

Comment:
This test requires drop inference (the next PR) and I could not figure out how to do this test without it